src/components/ContentPreview.tsx

```diff
@@ -31,61 +32,118 @@ const windowsMaxRenderSize = 10000000;
 
 interface ContentPreviewProps {
   resource: Resource;
+  isPreview?: boolean;
 }
 
-const ContentPreview: FC<ContentPreviewProps> = ({ resource }) => {
+const ContentPreview: FC<ContentPreviewProps> = ({ resource, isPreview = false }) => {
   const { t } = useTranslation();
-  const presentationMode = determinePresentationMode(resource);
-  const contentURL = `${API_URL}${API_PATHS.guiBackendResourcesContentPath}/${resource.contents.masterContent.identifier}/delivery?jwt=${localStorage.token}`;
+  const [defaultContent, setDefaultContent] = useState<Content | null>(null);
+  const [presentationMode, setPresentationMode] = useState<string>(
+    determinePresentationMode(resource.contents.masterContent)
+  );
+  const [isLoading, setLoading] = useState(false);
+  const UrlGeneratedFromMasterContent = `${API_URL}${API_PATHS.guiBackendResourcesContentPath}/${resource.contents.masterContent.identifier}/delivery?jwt=${localStorage.token}`;
+
+  useEffect(() => {
+    const fetch = async () => {
+      setLoading(true);
+      try {
+        const defaultContentResponse = await getResourceDefaultContent(resource.identifier);
+        setDefaultContent(defaultContentResponse.data);
+        setPresentationMode(determinePresentationMode(defaultContentResponse.data));
+      } catch (error) {
+        setDefaultContent(null);
+        setPresentationMode(determinePresentationMode(resource.contents.masterContent));
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    if (!isPreview) {
```

Comment:
ingen content hvis preview  ? Er ikke det litt strengt ? Er det mye styr å sjekke om forbåndsvisning er klar? Kunne vi ikke kjørt getResourceDefaultContent-kallet uansett og vist fram content om det var klart ?

Reply:
Content vises, så lenge det ikke er snakk om soundcloud. Problemet er at rett etter at man laster opp hovedfila eller legger inn en lenke så svarer getDefaultContent med feil mime-type for absolutt alt. De medfører at vi ikke får preview for noe som helst. Med denne sjekken så får vi ingen preview for soundcloud (og antageligvis 23video), fordi mime-type er feil, men alt annet vises frem som normalt.